### ISSUE TITLE
Feature request: convert bower.json.ignore array into npm equivalent

### ISSUE THREAD

**Issue by JakeChampion** [Financial-Times/origami-component-converter#33]
We have some origami components which have files that are not used for production such as: sketch files; test files; development configuration files.

It would be great to be able to ignore those files for the production artefacts that npm and bower create.

**Comment by JakeChampion** [comment]
We probably should not do this inside origami-component-converter because the filepaths to ignore are different for each origami component. E.G. o-grid we would ignore the `img` folder and o-ads we'd ignore `resources/images`.

If we did it in origami-component-converter then it could lead to a published npm version of a package not working even though the tests passed because the testing of the npm version does not test the artefact that would be published to npm. I.E. we do not do `npm pack` before running the tests.

**Comment by chee** [comment]
According to the bower spec it's always an array of strings, is that right?

So should we be echoing `bowerManifest.ignore.join("\n")` into `.npmignore`, (and also then catting `.gitignore` into `.npmignore` because the presence of an npmignore means no more gitignore).

> If we did it in origami-component-converter then it could lead to a published npm version of a package not working even though the tests passed because the testing of the npm version does not test the artefact that would be published to npm. I.E. we do not do npm pack before running the tests.

we can change this in `origami-ci`, and I think we should. I think the build would be better if we made a `build` directory for each build. One for the first `obt` build and one for the `occ`+`obt --ignore-bower` build. Then we also wouldn't need the `git clean`. And I think the `occ` tests should include running the tests on the published artifact.

alternately, we could delete all the files not listed in `files`, and all the files listed in the `.npmignore` before testing. but testing the actual tarball just sounds better huh

**Comment by JakeChampion** [comment]
>According to the bower spec it's always an array of strings, is that right?

Ah yes, array of strings, I read it as array or strings 🙈 -- https://github.com/bower/spec/blob/master/json.md#ignore

>we can change this in origami-ci, and I think we should. I think the build would be better if we made a build directory for each build. One for the first obt build and one for the occ+obt --ignore-bower build. Then we also wouldn't need the git clean. And I think the occ tests should include running the tests on the published artifact.

oh yes, being able to remove the `git clean` would be nice!

>but testing the actual tarball just sounds better huh

When I wrote about the `npm pack` and testing I also thought that is something we should be doing. Let's do it :-)

**Comment by chee** [comment]
I don't want to say this because it means updating all the CircleCI configs again but... for "1 directory for each build with no git clean"...
should those be two separate CircleCI tasks that can run in parallel?

**Comment by JakeChampion** [comment]
> I don't want to say this because it means updating all the CircleCI configs again but... for "1 directory for each build with no git clean"...
> should those be two separate CircleCI tasks that can run in parallel?

I think so, yes, it would make the builds faster overall which is really nice to have :)
